prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -954,37 +966,63 @@ class Hl7Serializer(val metadata: Metadata) : Logging {
         hl7Field: String,
         warnings: MutableList<String>
     ): String {
-        var dateTime = ""
+        var valueString = ""
         val fieldParts = getTerserSpec(hl7Field).split("-")
         if (fieldParts.size > 1) {
             val segment = terser.getSegment(fieldParts[0])
             val fieldNumber = fieldParts[1].toIntOrNull()
             if (segment != null && fieldNumber != null) {
-                val dtm = when (val value = segment.getField(fieldNumber, 0)) {
+                var dtm: Instant? = null
+                var rawValue = ""
+                when (val value = segment.getField(fieldNumber, 0)) {
                     // Timestamp
-                    is TS -> value.time
+                    is TS -> {
+                        dtm = value.time?.valueAsDate?.toInstant()
+                        rawValue = value.toString()
+                    }
                     // Date range. For getting a date time, use the start of the range
-                    is DR -> value.rangeStartDateTime?.time
-                    else -> null
+                    is DR -> {
+                        dtm = value.rangeStartDateTime?.time?.valueAsDate?.toInstant()
+                        rawValue = value.toString()
+                    }
+                    is DT -> {
+                        dtm = LocalDate.of(value.year, value.month, value.day)
+                            .atStartOfDay(ZoneId.systemDefault()).toInstant()
```

Comment:
Do we know what the system default is? Is it always UTC on Azure? I'm worried about the potential for time shift here.

Reply:
Yes, in Azure it is always UTC, however, for this conversion it does not matter as all it is doing is taking a date and converting it to something that has time for the sake of using other conversion methods in the next few lines.  It will not affect the date itself, so you can use any default timezone including EST in our locals.